packages/renderer/src/assets/download-file.ts

```diff
@@ -17,13 +19,48 @@ export const downloadFile = ({
 		  }) => void)
 		| undefined;
 }) => {
-	return new Promise<{sizeInBytes: number; to: string}>((resolve, reject) => {
+	return new Promise<Response>((resolve, reject) => {
 		let rejected = false;
+		let resolved = false;
+		let timeout: NodeJS.Timeout | undefined;
+
+		const resolveAndFlag = (val: Response) => {
+			resolved = true;
+			resolve(val);
+			if (timeout) {
+				clearTimeout(timeout);
+			}
+		};
+
 		const rejectAndFlag = (err: Error) => {
+			if (timeout) {
+				clearTimeout(timeout);
+			}
+
 			reject(err);
 			rejected = true;
 		};
 
+		const refreshTimeout = () => {
+			if (timeout) {
+				clearTimeout(timeout);
+			}
+
+			timeout = setTimeout(() => {
+				if (resolved) {
+					return;
+				}
+
+				rejectAndFlag(
+					new Error(
+						`Tried to download file ${url}, but no progress was made for 20 seconds`
+					)
+				);
+			}, 20000);
```

Comment:
I'm running `remotion render` from the CLI and using a Composition that loads a really large video file (2.54GB). This timeout error occurs every time. I had to revert back to a version before this was in place in order to render the video locally. Can you enable us to extend the timeout to more than 20 seconds?